code/datums/components/aquarium.dm

```diff
@@ -189,6 +189,8 @@
 
 /datum/component/aquarium/proc/on_click_alt(atom/movable/source, mob/living/user)
 	SIGNAL_HANDLER
+	if(!user.can_perform_action(src))
```

Comment:
Was this teated? I don't think this works because `base_click_alt()` already does this check for us

https://github.com/tgstation/tgstation/blob/b35e44c6f7200874e3d2a2f25065a02c316f6843/code/_onclick/click_alt.dm#L22

If you are trying to check for ghosts then a simple `isliving()` is all that's needed

Reply:
```js
/mob/dead/observer/can_perform_action(atom/movable/target, action_bitflags)
	return isAdminGhostAI(usr)
```

no this works